presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1393,11 +1402,26 @@ else if (expression instanceof LongLiteral) {
                     }
 
                     // otherwise, just use the expression as is
+                    boolean checkDeprecatedBehavior = false;
                     if (orderByExpression == null) {
                         orderByExpression = expression;
+                        checkDeprecatedBehavior = true;
                     }
 
                     ExpressionAnalysis expressionAnalysis = analyzeExpression(orderByExpression, sourceScope);
+
+                    if (checkDeprecatedBehavior) {
+                        for (Map.Entry<NodeRef<Expression>, FieldId> columnExpression : expressionAnalysis.getColumnReferences().entrySet()) {
+                            Optional<ResolvedField> resolvedField = orderByScope.tryResolveField(columnExpression.getKey().getNode());
+                            if (resolvedField.isPresent()) {
+                                if (!FieldId.from(resolvedField.get()).equals(columnExpression.getValue())) {
+                                    warningSink.logWarning(WarningCode.LEGACY_ORDER_BY, format("ORDER BY column %s was not resolved in accordance with the ANSI SQL specification, " +
```

Comment:
We should improve the message. Something like:

"Column %s referenced in ORDER BY expressions is present in SELECT clause aliases and source table. Your query may rely on behavior that doesn't conform to standard SQL semantics and will change in a future version. If the intent is to refer to a column in the source table, please qualify the reference with the appropriate table alias."